src/app/component/shared/header/header.component.spec.ts

```diff
@@ -68,443 +87,371 @@ describe("HeaderComponent", () => {
-    expect(link.attributes.getNamedItem("ng-reflect-router-link").value).toBe(
-      registerMenuItem.route.toString()
-    );
-  });
-
-  it("should display login link", () => {
-    spyOn(securityApi, "getAuthTrigger").and.callFake(
-      () => new BehaviorSubject(null)
-    );
-    fixture.detectChanges();
-
-    const link = fixture.nativeElement.querySelectorAll("a.nav-link")[4];
-    expect(link).toBeTruthy();
-    expect(link.innerText).toContain(loginMenuItem.label);
-    expect(link.attributes.getNamedItem("ng-reflect-router-link")).toBeTruthy();
-    expect(link.attributes.getNamedItem("ng-reflect-router-link").value).toBe(
-      loginMenuItem.route.toString()
-    );
-  });
-
-  it("should display profile name on logged in", fakeAsync(() => {
-    spyOn(securityApi, "isLoggedIn").and.callFake(() => {
-      return true;
-    });
-    spyOn(securityApi, "getAuthTrigger").and.callFake(
-      () => new BehaviorSubject(null)
-    );
-    spyOn(userApi, "show").and.callFake(() => {
-      const subject = new Subject<User>();
-
-      setTimeout(() => {
-        subject.next(
-          new User({
-            id: 1,
-            userName: "custom username",
-            rolesMask: 2,
-            rolesMaskNames: ["user"],
-            lastSeenAt: "2019-12-18T11:16:08.233+10:00"
-          })
+    ];
+
+    userRoles.forEach(userType => {
+      describe(userType.type + " user", () => {
+        let isLoggedIn: boolean;
+        let user: User;
+
+        beforeEach(() => {
+          if (userType.type === "guest") {
+            isLoggedIn = false;
+            user = undefined;
+          } else {
+            isLoggedIn = true;
+            user = new User({
+              id: 1,
+              userName: "Username",
+              rolesMask: userType.type === "admin" ? 1 : 2,
+              rolesMaskNames: userType.type === "user" ? ["user"] : ["admin"]
+            });
+          }
+        });
+
+        it("should create brand name link", () => {
+          setUser(isLoggedIn, user);
+          fixture.detectChanges();
+
+          const brand = fixture.nativeElement.querySelector("a.navbar-brand");
+          expect(brand).toBeTruthy();
+          expect(brand.innerText).toContain(env.values.brand.name);
+          assertRouterLink(brand, homeMenuItem.route.toString());
+        });
+
+        it("should create projects link", () => {
+          setUser(isLoggedIn, user);
+          fixture.detectChanges();
+
+          const link = fixture.nativeElement.querySelectorAll("a.nav-link")[0];
+          expect(link).toBeTruthy();
+          expect(link.innerText).toContain(projectsMenuItem.label);
+          assertRouterLink(link, projectsMenuItem.route.toString());
+        });
+
+        it("should create contact us link", () => {
+          setUser(isLoggedIn, user);
+          fixture.detectChanges();
+
+          const link = fixture.nativeElement.querySelectorAll("a.nav-link")[2];
+          expect(link).toBeTruthy();
```

Comment:
there are a lot of these `toBeTruthy`s lingering around your tests. Still a bit of a code smell